src/test/java/seedu/address/model/assignment/AssignmentListTest.java

```diff
@@ -138,4 +139,18 @@ public void resetData_withValidReadOnlyAddressBook_replacesData() {
         assertEquals(newData, assignments);
     }
 
+    @Test
+    public void reassignStudent_replacesData() {
+        AssignmentList assignments = getTypicalAssignmentList();
+        Student oldStudent = new StudentBuilder().withStudentId("A1234567I").build();
```

Comment:
Good that you have solved the bugs, would be better to solve the bugs in multiple commits/branches